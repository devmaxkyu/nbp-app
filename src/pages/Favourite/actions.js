/** 
 * @author Zemin W.
 * @description define actions to add and remove favorite currencies 
 * @lastModified 2019.10.4
 * */

const REMOVE_FAVOURITE = 'FAVOURITE_REMOVE'
const REMOVE_ALL_FAVOURITE = 'FAVOURITE_ALL_REMOVE'
const ADD_FAVOURITE = 'FAVOURITE_ADD'

export const actions = {
    REMOVE_FAVOURITE,
    REMOVE_ALL_FAVOURITE,
    ADD_FAVOURITE,
}

export const addFavourite = data => ({
    type: ADD_FAVOURITE,
    data
})

export const removeFavourite = data => ({
    type: REMOVE_FAVOURITE,
    data
})

export const removeAllFavourite = data =>({
    type: REMOVE_ALL_FAVOURITE,
    data
})