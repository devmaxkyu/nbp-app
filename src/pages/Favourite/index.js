/** 
 * @description page component to show favourite currencies
 * @lastModified 2019.10.4
 * @author zemin
 * */

import React from 'react'

class Favourite extends React.Component {
    render() {
        return (
            <h1>Hello, {this.props.name}</h1>
        )
    }
}

export default Favourite