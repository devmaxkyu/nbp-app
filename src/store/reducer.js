/** 
 * @author Zemin W.
 * @description combine reducers
 * @lastModified 2019.10.4
 * */

import { combineReducers } from 'redux'

import CurrencyDetailReducer from '../pages/CurrencyDetail/reducer'
import CurrencyListReducer from '../pages/CurrencyList/reducer'
import FavouriteReducer from '../pages/Favourite/reducer'

export default combineReducers({
    CurrencyDetailReducer,
    CurrencyListReducer,
    FavouriteReducer
})
