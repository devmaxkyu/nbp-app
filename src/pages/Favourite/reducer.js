/** 
 * @author Zemin W.
 * @description define reducers to refresh and filter currencies list
 * @lastModified 2019.10.4
 * */
import {createReducer} from 'redux-create-reducer'
import {actions} from './actions'

const initialState={
    list:[]
}

const reducer = createReducer(initialState, {
    [actions.ADD_FAVOURITE] (state, action){
        const list = state.list
        list.push(action.data)
        return { 
            list
        }
    },

    [actions.REMOVE_FAVOURITE] (state, action){
        const {code} = action.data
        const list = state.list.filter(item=> item.code !== code)
        return { 
            list
        }
    },

    [actions.REMOVE_ALL_FAVOURITE] (state, action){
        return { list:[]}
    }
   
})

export default reducer
