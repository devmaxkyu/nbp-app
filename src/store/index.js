/** 
 * @author Zemin W.
 * @description create store object
 * @lastModified 2019.10.4
 * */

import { createStore, applyMiddleware } from 'redux'
import { createLogger } from 'redux-logger'
import reducer from './reducer'

const store = createStore(reducer, applyMiddleware(createLogger()))

export default store
