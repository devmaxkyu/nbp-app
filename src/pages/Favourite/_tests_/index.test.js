
import React from 'react'
import Favourite from '../../Favourite'
import renderer from 'react-test-renderer'
import {Provider} from 'react-redux'
import store from '../../../store'

import { BrowserRouter as Router} from 'react-router-dom'

jest.mock('../../../services/nbpApi')

it('Favourite rendering', () => {
  const component = renderer.create(
    <Provider store = {store}><Router><Favourite/></Router></Provider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});