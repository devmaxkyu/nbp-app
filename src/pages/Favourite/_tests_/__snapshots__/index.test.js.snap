// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favourite rendering 1`] = `
<div>
  <h3>
    Favourite List
  </h3>
  <div
    className="App-btn-group"
  >
    <a
      className="App-link"
      href="#"
      onClick={[Function]}
    >
      All Remove Favourite
    </a>
  </div>
  <div
    className="table-wrap"
  >
    <table
      className="App-table"
    >
      <thead>
        <tr>
          <th>
            Currency Name
          </th>
          <th>
            Currency Value
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            colSpan="2"
          >
            No Favourite items, yay!
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
